src/main/java/com/epam/ta/reportportal/core/log/impl/GetLogHandlerImpl.java

```diff
@@ -205,6 +207,51 @@ private FilterCondition getLaunchCondition(Long launchId) {
 		return FilterCondition.builder().eq(CRITERIA_ITEM_LAUNCH_ID, String.valueOf(launchId)).build();
 	}
 
+	private void updateFilter(Filter filterable, String path) {
+		TestItem testItem = testItemRepository.findByPath(path)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.TEST_ITEM_NOT_FOUND, path));
+
+		updatePathCondition(testItem, filterable);
+
+		Launch launch = testItemService.getEffectiveLaunch(testItem);
+
+		FilterCondition.ConditionBuilder itemLaunchIdConditionBuilder = FilterCondition.builder()
+				.eq(CRITERIA_ITEM_LAUNCH_ID, String.valueOf(launch.getId()));
+
+		ConvertibleCondition launchIdCondition = ofNullable(testItem.getRetryOf()).map(retryOf -> (ConvertibleCondition) new CompositeFilterCondition(
+				Lists.newArrayList(itemLaunchIdConditionBuilder.withOperator(Operator.OR).build(),
+						FilterCondition.builder()
+								.eq(CRITERIA_RETRY_PARENT_LAUNCH_ID, String.valueOf(launch.getId()))
+								.withOperator(Operator.OR)
+								.build()
+				))).orElseGet(itemLaunchIdConditionBuilder::build);
+
+		filterable.getFilterConditions().add(launchIdCondition);
+
+	}
+
+	private void updatePathCondition(TestItem testItem, Filter filterable) {
```

Comment:
any comments describing what 'update' does?

Reply:
unit test for this

Reply:
updated. JUnit test provided

Reply:
https://github.com/reportportal/service-api/pull/1252/files#diff-89fa7202cc0ee99329af0a05299990aa